exec/vector/src/main/java/org/apache/drill/exec/vector/UntypedNullVector.java

```diff
@@ -0,0 +1,267 @@
+    final int actualLength = metadata.getBufferLength();
+    final int valueCount = metadata.getValueCount();
+    final int expectedLength = valueCount * VALUE_WIDTH;
+    assert actualLength == expectedLength : String.format("Expected to load %d bytes but actually loaded %d bytes", expectedLength, actualLength);
+
+    this.valueCount = valueCount;
+  }
+
+  @Override
+  public TransferPair getTransferPair(BufferAllocator allocator){
+    return new TransferImpl(getField(), allocator);
+  }
+
+  @Override
+  public TransferPair getTransferPair(String ref, BufferAllocator allocator){
+    return new TransferImpl(getField().withPath(ref), allocator);
+  }
+
+  @Override
+  public TransferPair makeTransferPair(ValueVector to) {
+    return new TransferImpl((UntypedNullVector) to);
+  }
+
+  public void transferTo(UntypedNullVector target){
+  }
+
+  public void splitAndTransferTo(int startIndex, int length, UntypedNullVector target) {
+  }
+
+  @Override
+  public int getPayloadByteCount() {
+    return 0;
+  }
+
+  private class TransferImpl implements TransferPair{
+    private UntypedNullVector to;
+
+    public TransferImpl(MaterializedField field, BufferAllocator allocator){
+      to = new UntypedNullVector(field, allocator);
+    }
+
+    public TransferImpl(UntypedNullVector to) {
+      this.to = to;
+    }
+
+    @Override
+    public UntypedNullVector getTo(){
+      return to;
+    }
+
+    @Override
+    public void transfer(){
+      transferTo(to);
+    }
+
+    @Override
+    public void splitAndTransfer(int startIndex, int length) {
+      checkBounds(startIndex);
+      checkBounds(startIndex + length - 1);
+      splitAndTransferTo(startIndex, length, to);
+    }
+
+    @Override
+    public void copyValueSafe(int fromIndex, int toIndex) {
+      checkBounds(fromIndex);
+      to.copyFromSafe(fromIndex, toIndex, UntypedNullVector.this);
+    }
+  }
+
+  public void copyFrom(int fromIndex, int thisIndex, UntypedNullVector from){
+  }
+
+  public void copyFromSafe(int fromIndex, int thisIndex, UntypedNullVector from){
+  }
+
+  private void checkBounds(int index) {
+    if (index < 0 || index >= valueCount) {
+      throw new IndexOutOfBoundsException(String.format(
+          "index: %d, expected: range(0, %d-1))", index, valueCount));
+    }
+  }
+  @Override
+  public void copyEntry(int toIndex, ValueVector from, int fromIndex) {
+    ((UntypedNullVector) from).data.getBytes(fromIndex * 4, data, toIndex * 4, 4);
+  }
+
+  public final class Accessor extends BaseAccessor {
+    @Override
+    public int getValueCount() {
+      return valueCount;
+    }
+
+    @Override
+    public boolean isNull(int index){
+      checkBounds(index);
+      return true;
+    }
+
+    public int isSet(int index) {
+      checkBounds(index);
+      return 0;
+    }
+
+    @Override
+    public Object getObject(int index) {
+      throw new UnsupportedOperationException();
```

Comment:
`return null`? Since the value is always null, shouldn't the Java Object value also be `null`?

Reply:
makes sense.  Changed to `return null`.